public/js/src/module/controller-webform.js

```diff
@@ -589,16 +598,21 @@ function _saveRecord(survey, draft, recordName, confirmed, errorMsg) {
         })
         .catch((error) => {
             console.error('save error', error);
-            errorMsg = error.message;
+            let errorMsg = error.message;
             if (
                 !errorMsg &&
                 error.target &&
                 error.target.error &&
                 error.target.error.name &&
                 error.target.error.name.toLowerCase() === 'constrainterror'
             ) {
-                errorMsg = t('confirm.save.existingerror');
-            } else if (!errorMsg) {
+                return _confirmRecordName(
+                    recordName,
+                    draft,
+                    t('confirm.save.existingerror')
+                ).then((name) => _saveRecord(survey, draft, name, true));
+            }
```

Comment:
Part 1 of the fix is to give users the ability to resolve this (rare, except for developers) issue.

Theoretically even if we could fix the indexedDb uniqueness perfectly, it is still possible for users (and even Enketo's auto-generated record names) to create a duplicate record name (draft or final). So this resolves that.